File: test/XRoadLib.Tests/Globals.fs
module Globals

open XRoadLib.Schema

let XRoadProtocol20 = XRoadLib.Tests.Contract.Configuration.CustomXRoad20Protocol.Instance
let XRoadProtocol31 = XRoadLib.Tests.Contract.Configuration.CustomXRoad31Protocol.Instance

let XRoadProtocol40 =
    let protocol = XRoadLib.Protocols.XRoad40Protocol("http://test-producer.x-road.eu/")
    protocol.SetContractAssembly(typeof<XRoadLib.Tests.Contract.Class1>.Assembly, null, 1u, 2u)
    protocol

type TestDefinition (typ) =
    interface IContentDefinition with
        member val UseXop = false
        member val RuntimeType = typ
        member val TypeName = null
        member val ArrayItemDefinition = null
        member val MergeContent = false

File: test/XRoadLib.Tests/Globals.cs
using XRoadLib.Headers;
using XRoadLib.Schema;
using XRoadLib.Tests.Contract;

namespace XRoadLib.Tests;

public static class Globals
{
    public static ServiceManager<XRoadHeader> ServiceManager { get; } = new("4.0", new DefaultSchemaExporter("http://test-producer.x-road.eu/", typeof(Class1).Assembly, new[] { 1u, 2u, 3u }));

    public static ContentDefinition GetTestDefinition(Type type)
    {
        return new TestDefinition(type);
    }

    private sealed class TestDefinition : ContentDefinition
    {
        public TestDefinition(Type type)
            : base(new TestParticleDefinition())
        {
            RuntimeType = type;
        }
    }

    private sealed class TestParticleDefinition : ParticleDefinition
    { }
}